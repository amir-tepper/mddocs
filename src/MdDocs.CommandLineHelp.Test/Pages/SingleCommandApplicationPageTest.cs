using ApprovalTests;
using ApprovalTests.Reporters;
using Grynwald.MarkdownGenerator;
using Grynwald.MdDocs.CommandLineHelp.Model;
using Grynwald.MdDocs.CommandLineHelp.Pages;
using Xunit;

namespace Grynwald.MdDocs.CommandLineHelp.Test.Pages
{
    [Trait("Category", "SkipWhenLiveUnitTesting")]
    [UseReporter(typeof(DiffReporter))]
    public class SingleCommandApplicationPageTest
    {
        [Fact]
        public void GetDocument_returns_expected_document_01()
        {
            var parameters = new UnnamedCommandDocumentation(
                application: new TestAppDocumentation(),
                options: new[]
                {
                    new OptionDocumentation("parameter1", required: true),
                    new OptionDocumentation("parameter2", 'x', required: true),
                    new OptionDocumentation(null, 'y', required: true),
                },
                values: new[]
                {
                    new ValueDocumentation(0),
                    new ValueDocumentation(1, name: "Value2", required: true),
                    new ValueDocumentation(2, name: "Value3", helpText: "Help text for value 3"),
                    new ValueDocumentation(3, name: "Value4", hidden: true),
                });

            var application = new SingleCommandApplicationDocumentation(name: "ApplicationName", parameters, "1.2.3", new[] { "Usage line 1", "Usage line2" });
            Approve(application);
        }


        private void Approve(SingleCommandApplicationDocumentation model)
        {
            var pathProvider = new DefaultCommandLineHelpPathProvider();
            var documentSet = new DocumentSet<IDocument>();

            var applicationPage = new SingleCommandApplicationPage(documentSet, pathProvider, model);
            documentSet.Add(pathProvider.GetPath(model), applicationPage);

            var doc = applicationPage.GetDocument();

            Assert.NotNull(doc);
            var writer = new ApprovalTextWriter(doc.ToString());
            Approvals.Verify(writer, new ApprovalNamer(), Approvals.GetReporter());
        }
    }
}
