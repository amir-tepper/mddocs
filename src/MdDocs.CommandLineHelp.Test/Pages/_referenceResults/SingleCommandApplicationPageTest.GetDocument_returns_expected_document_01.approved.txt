<!--  
 =================================================================   
   Auto-Generated:   
   The contents of this file were generated by a tool.  
   Changes to this file may be list if the file is regenerated  
 =================================================================   
-->

# ApplicationName Command Line Reference

**Version:** 1.2.3

## Usage

Usage line 1  
Usage line2

## Usage

```
TestApp [<VALUE>]
        <Value2>
        [<Value3>]
        --parameter1 <VALUE>
        --parameter2|-x <VALUE>
        -y <VALUE>
```

## Parameters

| Position                          | Name                                     | Short Name                 | Required | Description           |
| --------------------------------- | ---------------------------------------- | -------------------------- | -------- | --------------------- |
| [0](#parameter-position-0)        |                                          |                            | No       |                       |
| [1](#value2-parameter-position-1) | [*Value2*](#value2-parameter-position-1) |                            | Yes      |                       |
| [2](#value3-parameter-position-2) | [*Value3*](#value3-parameter-position-2) |                            | No       | Help text for value 3 |
|                                   | [parameter1](#parameter1-parameter)      |                            | Yes      |                       |
|                                   | [parameter2](#parameter2-parameter)      | [x](#parameter2-parameter) | Yes      |                       |
|                                   |                                          | [y](#y-parameter)          | Yes      |                       |

### Parameter (Position 0)

|                |        |
| -------------- | ------ |
| Position:      | 0      |
| Required:      | No     |
| Default value: | *None* |

___

### `Value2` Parameter (Position 1)

|                |        |
| -------------- | ------ |
| *Name:*        | Value2 |
| Position:      | 1      |
| Required:      | Yes    |
| Default value: | *None* |

___

### `Value3` Parameter (Position 2)

Help text for value 3

|                |        |
| -------------- | ------ |
| *Name:*        | Value3 |
| Position:      | 2      |
| Required:      | No     |
| Default value: | *None* |

___

### `parameter1` Parameter

|                |                   |
| -------------- | ----------------- |
| Name:          | parameter1        |
| Position:      | *Named parameter* |
| Required:      | Yes               |
| Default value: | *None*            |

___

### `parameter2` Parameter

|                |                   |
| -------------- | ----------------- |
| Name:          | parameter2        |
| Short name:    | x                 |
| Position:      | *Named parameter* |
| Required:      | Yes               |
| Default value: | *None*            |

___

### `y` Parameter

|                |                   |
| -------------- | ----------------- |
| Name:          | y                 |
| Position:      | *Named parameter* |
| Required:      | Yes               |
| Default value: | *None*            |

___

*Documentation generated by [MdDocs](https://github.com/ap0llo/mddocs)*
